pydicom/filewriter.py

```diff
@@ -486,13 +486,16 @@ def write_data_element(fp, data_element, encodings=None):
     # valid pixel data with undefined length shall contain encapsulated
     # data, e.g. sequence items - raise ValueError otherwise (see #238)
     if is_undefined_length and data_element.tag == 0x7fe00010:
-        val = data_element.value
-        if (fp.is_little_endian and not
-                val.startswith(b'\xfe\xff\x00\xe0') or
-                not fp.is_little_endian and
-                not val.startswith(b'\xff\xfe\xe0\x00')):
-            raise ValueError('Pixel Data with undefined length must '
-                             'start with an item tag')
+        encap_item = b'\xfe\xff\x00\xe0'
+        if not fp.is_little_endian:
+            # Non-conformant endianness
+            encap_item = b'\xff\xfe\xe0\x00'
```

Comment:
Should we issue a warning in this case?

Reply:
I don't think we need to, none of the compressed syntaxes are BE so the only way to get here would be by avoiding `Dataset.save_as` (because with Big Endian Explicit VR `is_undefined_length` would be `False`) which implies they know what they're doing.

Reply:
Thanks, agreed.